src/IonBinaryReader.ts

```diff
@@ -62,194 +62,156 @@ const TB_LIST_CLOSE    = 22;
-    t._parser.stepOut();
-    t._raw_type = BOC;
-  }
-
-  valueType() : IonType {
-    return get_ion_type(this._raw_type);
-  }
-
-  depth() : number {
-    return this._parser.depth();
-  }
-
-  fieldName() : string {
-    let t: BinaryReader = this;
-    var n, s;
-    n = t._parser.getFieldId();
-    s = this.getSymbolString(n)
-    return s;
-  }
-
-  hasAnnotations() : boolean {
-    return this._parser.hasAnnotations();
-  }
-
-  annotations() : string[] {//TODO binary support
-      return ["test"];
-  }
-
-  getAnnotation(index: number) : string {
-    let t: BinaryReader = this;
-    var id, n;
-    id = t._parser.getAnnotation(index);
-    n = this.getSymbolString(id);
-    return n;
-  }
-
-  isNull() : boolean {
-    let t: BinaryReader = this;
-    var is_null = (t._raw_type === TB_NULL) || t._parser.isNull();
-    return is_null;
-  }
-
-  stringValue() : string {
-    let t: BinaryReader = this;
-    var n, s, p = t._parser;
-    if (t.isNull()) {
-      s = "null";
-      if (t._raw_type != TB_NULL) {
-        s += "." + get_ion_type(t._raw_type).name;
-      }
-    }
-    else if (get_ion_type(t._raw_type).scalar) {
-      // BLOB is a scalar by you don't want to just use the string 
-      // value otherwise all other scalars are fine as is
-      if (t._raw_type === TB_SYMBOL) {
-        n = p.numberValue();
-        s = this.getSymbolString(n);
-      }
-      else {
-        s = p.stringValue();
-      }
-    }
-    return s;
-  }
-
-  numberValue() : number {
-    return this._parser.numberValue();
-  }
-
-  byteValue() : number[] {
-    return this._parser.byteValue();
-  }
-
-  ionValue() : never {
-    throw new Error("E_NOT_IMPL: ionValue");
-  }
-
-  booleanValue() : boolean {
-    return this._parser.booleanValue();
-  }
-
-  decimalValue() : Decimal {
-    return this._parser.decimalValue();
-  }
-
-  timestampValue() : Timestamp {
-    return this._parser.timestampValue();
-  }
-
-  value() : any {
-    switch(this.valueType()) {
-      case IonTypes.BLOB:
-      case IonTypes.CLOB:
-        return this.byteValue();
-      case IonTypes.BOOL:
-        return this.booleanValue();
-      case IonTypes.DECIMAL:
-        return this.decimalValue();
-      case IonTypes.FLOAT:
-      case IonTypes.INT:
-        return this.numberValue();
-      case IonTypes.STRING:
-      case IonTypes.SYMBOL:
-        return this.stringValue();
-      case IonTypes.TIMESTAMP:
-        return this.timestampValue();
-      default:
-        return undefined;
-    }
-  }
-
-  private getSymbolString(symbolId: number) : string {
-    let s: string = undefined;
-    if (symbolId > 0) {
-      s = this._symtab.getSymbol(symbolId);
-      if (typeof(s) == 'undefined') {
-        s = "$" + symbolId.toString();
-      }
-    }
-    return s;
-  }
+    private _parser: ParserBinaryRaw;
+    private _cat: Catalog;
+    private _symtab: LocalSymbolTable;
+    private _type: number;
+//where do shared symbol tables fall into place here?
```

Comment:
The reader will encounter shared symbol table imports in the stream and look them up in the provided catalog.